### PR TITLE
Add more `make` targets: `clean/no-push-down/push-down-without-v…

### PR DESCRIPTION
**Pull request by iosmanthus**
This pull requests refactor the `copr-test` inorder to make reproducing a failed SQL more conveniently.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,8 +5,6 @@ executes TiDB push-down executors to improve performance.
 
 ## Push Down Test
 
-`./push-down-test`
```
Comment (breeswish) [inline README.md]: revert

File: README.md [hunk 2]
```diff
@@ -36,7 +34,7 @@ TiDB side for now.
 ### Test Cases
 
 We have already added some test cases generated by [randgen] in the push-down-test. Feel free to
```
Comment (breeswish) [inline README.md]: revert

File: README.md [hunk 3]
```diff
@@ -66,14 +64,23 @@ cd ../copr-test
...
+make push-down-without-vec
+
+# If you want to clean the `build` directory and kill all the tidb/pd/tikv processes, run:
+make clean
 ```
 
 If you want to filter some test cases:
 
 ```
-include=1_arith_1.sql make push-down-test
```
Comment (breeswish) [inline README.md]: revert

File: push-down-test/.gitignore [hunk 4]
```diff
@@ -1 +0,0 @@
-/bin/
```
Comment (breeswish) [inline push-down-test/.gitignore]: revert

File: push-down-test/run-test.sh [hunk 5]
```diff
@@ -0,0 +1,366 @@
+# GO environment variables
```
Comment (breeswish) [inline push-down-test/run-test.sh]: revert the rename so that github diff will work

Comment (lonng) [reply push-down-test/run-test.sh]: Agreed

File: README.md [hunk 6]
```diff
@@ -36,7 +34,7 @@ TiDB side for now.
 ### Test Cases
 
 We have already added some test cases generated by [randgen] in the push-down-test. Feel free to
-add new ones. Your test case should be placed in the `push-down-test/sql` directory and ends with
+add new ones. Your test case should be placed in the `sql` directory and ends with
```
Comment (breeswish) [inline README.md]: revert?

File: .gitignore [hunk 7]
```diff
@@ -3,7 +3,7 @@
 *.swp
 .DS_Store
 
-*.log
+**.log
```
Comment (breeswish) [inline .gitignore]: `**`?

File: push-down-test/.gitignore (hunk 4 above)
Comment (iosmanthus) [reply push-down-test/.gitignore]: move to `push-down-test/build`

File: README.md (hunk 6 above)
Comment (iosmanthus) [reply README.md]: Done

File: push-down-test/run-tests.sh [hunk 8]
```diff
@@ -1,240 +1,365 @@
...
   echo
-  echo "+ Starting TiKV"
-  "$tikv_bin" -C "$push_down_no_batch_config_dir"/tikv.toml --log-file "$tikv_no_batch_log_file" -L ${log_level} &
-  tikv_no_batch_pid=$!
+  echo "+ Launching TiKV for $5 test"
+  $1 -C $2 --log-file $3 -L $4 &
 
-  "$tikv_bin" -C "$push_down_with_batch_config_dir"/tikv.toml --log-file "$tikv_with_batch_log_file" -L ${log_level} &
-  tikv_with_batch_pid=$!
+  # Return the PID of the new PD process
```
Comment (lonng) [inline push-down-test/run-tests.sh]:
```suggestion
  # Return the PID of the new TiKV process
```

File: push-down-test/run-tests.sh [hunk 9]
```diff
@@ -1,240 +1,365 @@
...
-  "$tidb_bin" -log-file "$push_down_no_batch_tidb_log_file" -config "$push_down_no_batch_config_dir"/tidb.toml -L ${log_level} &
-  tidb_push_down_no_batch_pid=$!
+  echo "+ Launching TiDB for $6 test"
+  export GO_FAILPOINTS="$5"
+  $1 -config $2 -log-file $3 -L $4 &
 
-  export GO_FAILPOINTS="github.com/pingcap/tidb/expression/PushDownTestSwitcher=return(\"$push_down_func_list\");github.com/pingcap/tidb/expression/PanicIfPbCodeUnspecified=return(true)"
-  "$tidb_bin" -log-file "$push_down_with_batch_tidb_log_file" -config "$push_down_with_batch_config_dir"/tidb.toml -L ${log_level} &
-  tidb_push_down_with_batch_pid=$!
+  # Return the PID of the new PD process
```
Comment (lonng) [inline push-down-test/run-tests.sh]: ditto

File: push-down-test/run-tests.sh [hunk 10]
```diff
@@ -1,240 +1,365 @@
...
+  make -C ${tidb_src_dir}
+
+  tidb_bin=${tidb_src_dir}/bin/tidb-server
 }
 
 function build_tester() {
   echo
-  echo "+ Building Push Down Tester"
-  echo "  - Building from ${push_down_test_bin}"
+  echo "+Building Push Down Tester"
```
Comment (lonng) [inline push-down-test/run-tests.sh]:
```suggestion
  echo "+ Building Push Down Tester"
```

File: push-down-test/run-tests.sh [hunk 11]
```diff
@@ -1,240 +1,365 @@
...
+    -include "${include}" \
+    -exclude "${exclude}"
+  readonly exit_code=$?
+
+  echo "+ Test finished"
+  echo "  - ${push_down_test_bin} exit code is ${exit_code}"
+
+  clean_all_proc
+  exit $exit_code
+
```
Comment (lonng) [inline push-down-test/run-tests.sh]: Remove this empty line.

File: push-down-test/run-tests.sh (hunk 8 above)
Comment (iosmanthus) [reply push-down-test/run-tests.sh]: Done

File: push-down-test/run-tests.sh (hunk 9 above)
Comment (iosmanthus) [reply push-down-test/run-tests.sh]: Done

File: push-down-test/run-tests.sh (hunk 10 above)
Comment (iosmanthus) [reply push-down-test/run-tests.sh]: Done

File: push-down-test/run-tests.sh (hunk 11 above)
Comment (iosmanthus) [reply push-down-test/run-tests.sh]: Done

**Comment by iosmanthus** [comment]
@lonng PTAL

**Comment by lonng** [comment]
@iosmanthus The prepare step has executed in the run_tests.sh scripts. It's better to remove the prepare logic in `main.go`

**Pull request merged**